src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -303,6 +303,13 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::Duplicate
         duplicated->##settingName(source.##settingName());                                                                                     \
     }
 
+#define DUPLICATE_FONT_SETTING_MACRO(settingName)                                                                                                                                    \
+    if (source.FontInfo().Has##settingName() ||                                                                                                                                      \
+        (source.FontInfo().##settingName##OverrideSource() != nullptr && source.FontInfo().##settingName##OverrideSource().SourceProfile().Origin() != OriginTag::ProfilesDefaults)) \
+    {                                                                                                                                                                                \
+        duplicated->FontInfo().##settingName(source.FontInfo().##settingName());                                                                                                     \
+    }

```

Comment:
BTW In classic C you often write:
```cpp
do {
    if (...) break;
    if (...) break;
    if (...) break;
    do_something();
} while (0)
```

It would allow you to prevent the repeated COM calls and reduce the line width.

Reply:
I'd like to keep it this way for consistency with `DUPLICATE_SETTING_MACRO` and `DUPLICATE_APPEARANCE_SETTING_MACRO`!

Reply:
we should book work to fix this